### ISSUE TITLE
MultipartFile parameter must not be added as template variable

### ISSUE THREAD

**Issue by reda-alaoui** [spring-projects/spring-hateoas#1776]
I use Spring HATEOAS 1.3.4.

Given:

```java
@Controller
@RequestMapping("/foo")
public class MyController {
    
    @GetMapping
    public ResponseEntity<?> get() {
       return ResponseEntity.ok(
         new RepresentationModel<>(
           linkTo(methodOn(MyController.class).get())
             .withSelfRel()
             .andAffordance(afford(methodOn(MyController.class).uploadFile(null)))
         )
       );
    }

    @PostMapping(value = "/file", consumes = "multipart/form-data")
    public ResponseEntity<?> uploadFile(@RequestParam("file") MultipartFile file) {
      return ResponseEntity.noContent().build();
    }

}
```

Performing:

```
GET http://localhost:8080/foo/
```

Produces:
```json
{
  "_links": {
    "self": {
      "href": "http://localhost:8080/api/foo"
    }
  },
  "_templates": {
    "default": {
      "method": "put",
      "contentType": "multipart/form-data",
      "properties": [],
      "target": "http://localhost:8080/api/foo/file?file={file}"
    }
  }
}
```

As you can see, the `target` property is not valid because of `file` parameter.

**Comment by odrotbohm** [comment]
Good catch! Thanks for the PR, too. 1.3 is unfortunately out of maintenance. Could you upgrade to a more recent 1.4.x?

**Comment by reda-alaoui** [comment]
We were already using a fork of `1.3.4` containing #1312 . I added #1777 to our fork.
So it's ok for us if it is only merged on `1.4`.

But it would be really nice if someone could take a look at #1312 :blush: :pray:

**Comment by odrotbohm** [comment]
I've merged and back-ported this into 1.5 and 1.4. I've slightly tweaked the implementation to also skip the `MultipartFile` instances even if non-`null` values are handed into the dummy method invocation.
